nixos/modules/programs/onedrive.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.onedrive;
+
+	onedriveLauncher =  pkgs.writeShellScriptBin
+		"onedrive-launcher"
+		''
+			# XDG_CONFIG_HOME is not recognized in the environment here.
+			if [ -f $HOME/.config/onedrive-launcher ]
```

Comment:
Is this a commonly used file name used for this?

If not, maybe this should be an option that uses this name as the default?

Better still, how about using an option that allows you to list config directories? Maybe `services.onedrive.configDirs` with the default value being `[ "~/.config/onedrive" ]`?

Reply:
I am not too keen on doing things at the "system" level, this is thing that should be done at user level. 

Basically the user does what he always does- initialize with oauth and config, and then the service runs as is. Except this time, you can have multiple accounts.

The filename- "onedrive-launcher" is not standard. This is our creation. Once again, it's per user and hence probably good to keep standard and also not customizable per system (if it were customizable per user that would have been fine)

Reply:
Gotcha, makes sense.